src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_connection/manage_direct_query_data_connections_table.tsx

```diff
@@ -3,145 +3,215 @@
+  deleteMultipleDataSources,
+  getDefaultDataSourceId,
+  setFirstDataSourceAsDefault,
+} from '../../utils';
+import { LoadingMask } from '../../loading_mask';
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
 
-export const ManageDirectQueryDataConnectionsTable: React.FC<ManageDirectQueryDataConnectionsTableProps> = ({
-  http,
-  notifications,
-  savedObjects,
-  uiSettings,
-  featureFlagStatus,
-  application,
-}) => {
+export const ManageDirectQueryDataConnectionsTable = ({ history }: RouteComponentProps) => {
+  const { savedObjects, http, notifications, uiSettings } = useOpenSearchDashboards<
+    DataSourceManagementContext
+  >().services;
   const [observabilityDashboardsExists, setObservabilityDashboardsExists] = useState(false);
   const [showIntegrationsFlyout, setShowIntegrationsFlyout] = useState(false);
   const [integrationsFlyout, setIntegrationsFlyout] = useState<React.JSX.Element | null>(null);
 
-  const [data, setData] = useState<DataConnection[]>([]);
+  const [data, setData] = useState<DataSourceTableItem[]>([]);
   const [isModalVisible, setIsModalVisible] = useState(false);
-  const [modalLayout, setModalLayout] = useState(<EuiOverlayMask />);
-  const [selectedConnection, setSelectedConnection] = useState<string | undefined>(undefined);
-  const [selectedDataSourceId, setSelectedDataSourceId] = useState<string | undefined>(undefined);
-  const [searchText, setSearchText] = useState<string>('');
+  const [isDeleting, setIsDeleting] = React.useState<boolean>(false);
+  const [selectedDataSources, setSelectedDataSources] = useState<DataSourceTableItem[]>([]);
+
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<
+    Record<string, React.ReactNode>
+  >({});
   const [isLoading, setIsLoading] = useState<boolean>(false);
-  const history = useHistory();
 
-  const fetchDataSources = useCallback(() => {
-    if (featureFlagStatus && selectedDataSourceId === undefined) return;
+  /* Table selection handlers */
+  const onSelectionChange = (selected: DataSourceTableItem[]) => {
+    setSelectedDataSources(selected);
+  };
 
-    const endpoint =
-      featureFlagStatus && selectedDataSourceId !== undefined
-        ? `${DATACONNECTIONS_BASE}/dataSourceMDSId=${selectedDataSourceId}`
-        : `${DATACONNECTIONS_BASE}`;
+  const selection = {
+    onSelectionChange,
+  };
 
-    setIsLoading(true);
+  const setDefaultDataSource = async () => {
+    try {
+      for (const dataSource of selectedDataSources) {
+        if (getDefaultDataSourceId(uiSettings) === dataSource.id) {
+          await setFirstDataSourceAsDefault(savedObjects.client, uiSettings, true);
+        }
+      }
+    } catch (e) {
+      notifications.toasts.addDanger(
+        i18n.translate('dataSourcesManagement.directQueryTable.setDefaultDataSourceFailMsg', {
+          defaultMessage:
+            'Unable to find a default datasource. Please set a new default datasource.',
```

Comment:
This message sounds a bit off. How about:
```suggestion
            'No default data source has been set. Please select a new default data source.',
```